step_input.go

```diff
@@ -27,6 +29,18 @@ func (s *InputStep) MarshalJSON() ([]byte, error) {
 	return json.Marshal(s.Contents)
 }
 
+// MarshalYAML returns s.Scalar if it's not empty, otherwise s.Contents if that
```

Comment:
might it be useful to include a func like:
```Go
func (s *InputStep) marshalValue() (any, error) { // name tbd
	if s.Scalar != "" {
		return s.Scalar
	}

	if len(s.Contents) == 0 {
		return nil, errors.New("empty input step")
	}

	return s.Contents
}
```

and then include that in both `MarshalYAML` and `MarshalJSON`?
```Go
func (s *InputStep) MarshalYAML() (any, error) { return yaml.Marshal(s.marshalValue()) }
func (s *InputStep) MarshalJSON() (any, error) { return json.Marshal(s.marshalValue()) }

```

i'm very happy to merge as is, in writing this out it feels like it might be overly-DRY